adoc/chapters/programming_interface.adoc

```diff
@@ -19596,15 +19596,14 @@ template <typename T> const stream& operator<<(const stream& os, const T& rhs)
 
 
 
-=== Synchronization
+=== Output
 
-An instance of the SYCL [code]#stream# class is required to synchronize with the host, and must output
-everything that is streamed to it via the [code]#operator<<()# operator before a flush operation (that
-doesn't exceed the [code]#workItemBufferSize# or [code]#totalBufferSize# limits) within a SYCL
-kernel function by the time that the event associated with a command group submission enters the completed
-state. The point at which this synchronization occurs and the member function by which this synchronization is
-performed are implementation-defined. For example it is valid for an implementation to use
-[code]#printf()#.
```

Comment:
Why remove all the `printf`  and the implementation detail stuff?

Reply:
The sentence before this one says that:

> An instance of the SYCL `stream` class is required to output everything ... by the time that the event associated with a command group submission enters the completed state.

If we reworded the removed sentence to replace "synchronization" with "output", it would read:

> The point at which this output occurs and the member function by which this output is performed are implementation-defined.

...which just feels like repetition to me.  We've already said that output occurs at some point before the event enters the completed state.

I don't think the reference to `printf` adds anything.  (It's also confusing, because `printf` isn't a member function.  So either `printf` isn't a valid example of what the sentence before it is trying to describe, or the sentence before it is wrong.)

In hindsight, I should probably have separated these sorts of edits from the synchronization re-wording.  But it was too tempting to try and simplify things given that I was editing the sentences already.

Reply:
>In hindsight, I should probably have separated these sorts of edits from the synchronization re-wording. But it was too tempting to try and simplify things given that I was editing the sentences already.

:D I know the feeling! Your justification makes sense! I just wanted to double those tangential changes :) 

Can we replace it with something like:

>The point at which the flush is performed is implementation-defined

I agree with the removal of `printf` which was unclear to me too. Hence my original residence to removing it. Removing stuff I don't understand seam... wrong)

Reply:
I've restored the note about the implementation-defined behavior in aa30f4da953f806eeee247287b65f09d4fc89084.